### ISSUE TITLE
Go to definition opens new tab even if target file is already open in another panel

### ISSUE THREAD

**Issue by N8th8n8el** [zed-industries/community#954]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Steps to reproude (for me this is a typescript project, but don't think it matters):
- Have file `A` which contains the definition for something that is used in file `B`.
- Have file `A` open on the left and file `B` on the right in a split panel configuration.
- Click on `Go to definition` on the thing in file `B`.
-> Opens up file `A` in the right panel instead of focusing the already open file on the left.

I find this a bit annoying when I, for example, have code open on the left and a test file on the right.

### Environment

Zed: v0.72.5 (stable)
OS: macOS 13.1.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by N8th8n8el** [comment]
I guess it is debatable whether this is a bug or a feature request.

**Comment by aexvir** [comment]
intellij based ides like goland have it for sure
![capture-20230214162352](https://user-images.githubusercontent.com/8055505/218781397-62b83f06-e11b-4059-9596-f0cd1b7f3e97.png)

**Comment by JosephTLyons** [comment]
> intellij based ides like goland have it for sure
> ![capture-20230214162352](https://user-images.githubusercontent.com/8055505/218781397-62b83f06-e11b-4059-9596-f0cd1b7f3e97.png)

Yeah, this makes sense to include it as a setting.  If we added this behavior, we'd likely put it behind a setting as well.

**Comment by N8th8n8el** [comment]
> Hey @N8th8n8el, do you use another editor that has this behavior? I believe VS Code, by default, does what we are doing right now.

>intellij based ides like goland have it for sure

Exactly, I used `WebStorm` before.
